src/main/java/com/google/devtools/build/lib/authandtls/credentialhelper/CredentialHelperProvider.java

```diff
@@ -129,6 +129,23 @@ private void checkHelper(Path path) throws IOException {
           path.isExecutable(), "Credential helper %s is not executable", path);
     }
 
+    /**
+     * Adds a credential helper to use for all {@link URI}s matching the provided pattern, or as
+     * default credential helper if {@code pattern} is empty.
+     *
+     * <p>See {@link #add(String, Path)} for the syntax of {@code pattern}.
+     */
+    public Builder add(Optional<String> pattern, Path helper) throws IOException {
```

Comment:
IMO it's much clearer for this method to receive a non-optional `pattern`, and call `add(Path)` directly in the case of a default helper.